src/runtime/auto-prompt-manager.ts

```diff
@@ -253,13 +273,24 @@ export class AutoPromptManager {
         clientConfig.autoPromptConfig
       ));
 
-    const potentialAction = await this.getAction_({
-      article,
-      autoPromptType,
-      dismissedPrompts,
-      canDisplayMonetizationPrompt,
-      shouldShowMonetizationPromptAsSoftPaywall,
-    });
+    let potentialAction: Intervention | void;
+
+    // Frequency Capping Flow
+    if (this.promptFrequencyCappingEnabled_) {
```

Comment:
Separated code paths for new frequency cap flow.

For 2, this change is a bit more involved because of the existing checks for the mini prompt. Refactoring this would affect a lot of the tests, as well. I am inclined to leave this as is for now and clean this up further after the legacy paths are deprecated and mini prompt api is fully deprecated. Does that sound ok with you?